AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt

```diff
@@ -5,30 +5,112 @@ import android.view.LayoutInflater
+    private lateinit var editTextSurname: EditText
+    private lateinit var editTextEMail: EditText
+    private lateinit var editTextAddress: EditText
+    private lateinit var editTextSocialSecurityID: EditText
+    private lateinit var editTextPhoneNumber: EditText
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
         return layoutInflater.inflate(R.layout.fragment_profile, null)
     }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         initFields(view)
+        initListeners()
+        assignValues()
+
+    }
+
+    private fun assignValues() {
+        val user = PreferencesRepo.getUser(requireContext())
+        if (user != null) {
+            editTextName.setText(user.name)
+            editTextSurname.setText(user.surname)
+            editTextEMail.setText(user.email)
+            editTextAddress.setText(user.address)
+            editTextSocialSecurityID.setText(user.secid)
+            editTextPhoneNumber.setText(user.phonenumber)
+        }
+
+    }
 
+    private fun initListeners() {
         btnlogout.setOnClickListener {
             PreferencesRepo.deleteUser(requireContext())
             PreferencesRepo.deleteVaccination(requireContext())
             findNavController().navigate(R.id.action_profileFragment_to_loginFragment)
+
+        }
+
+        btnedit.setOnClickListener {
+            val oldUser = PreferencesRepo.getUser(requireContext())
+            if (oldUser != null) {
+                val user = User(
+                    oldUser.id,
+                    editTextName.text.toString(),
+                    editTextSurname.text.toString(),
+                    oldUser.email,
+                    editTextAddress.text.toString(),
+                    editTextSocialSecurityID.text.toString(),
+                    editTextPhoneNumber.text.toString(),
+                    oldUser.password
+                )
+                PreferencesRepo.saveUser(requireContext(), user)
+
+                FirebaseRepo.updateUser(user, object : FirebaseListener {
+                    override fun onSuccess(user: User?) {
+                        btnedit.isEnabled = true
+                    }
+
+                    override fun onStart() {
+                        btnedit.isEnabled = false
+                    }
+
+                    override fun onFailure() {
+                        btnedit.isEnabled = true
+                        Toast.makeText(
+                            requireContext(),
+                            getString(R.string.error_firebase_communication),
+                            Toast.LENGTH_LONG
+                        ).show()
+                    }
+                })
```

Comment:
Please add some notification to the user so that he knows whats going on (fail, success ...)

Reply:
Fixed with our new commit